framework/include/postprocessors/ComparisonPostprocessor.h

```diff
@@ -0,0 +1,71 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef COMPARISONPOSTPROCESSOR_H
+#define COMPARISONPOSTPROCESSOR_H
+
+#include "GeneralPostprocessor.h"
+
+class ComparisonPostprocessor;
+
+template <>
+InputParameters validParams<ComparisonPostprocessor>();
+
+/**
+ * Base class for comparing quantities and producing a boolean value
+ *
+ * This post-processor implements a number of different comparisons between
+ * vector post-processor \c a and vector post-processor \c b:
+ * \li <b>equals</b>: <tt>a == b</tt>?
+ * \li <b>greater_than</b>: <tt>a > b</tt>?
+ * \li <b>greater_than_equals</b>: <tt>a >= b</tt>?
+ * \li <b>less_than</b>: <tt>a < b</tt>?
+ * \li <b>less_than_equals</b>: <tt>a <= b</tt>?
+ *
+ * For all comparisons, a "fuzzy" comparison is made via the corresponding
+ * function in \c MooseUtils.
+ */
+class ComparisonPostprocessor : public GeneralPostprocessor
+{
+public:
+  ComparisonPostprocessor(const InputParameters & parameters);
+
+protected:
+  /**
+   * Performs the selected comparison on the two values
+   *
+   * @param[in] a   First value in the comparison
+   * @param[in] b   Second value in the comparison
+   *
+   * @return Boolean value for the comparison being true
+   */
+  bool comparisonIsTrue(const Real & a, const Real & b) const;
+
+  /**
+   * Converts a boolean value to a real value, according to conventions
+   *
+   * @param[in] boolean_value   Boolean value
+   *
+   * @return Real value corresponding to "true" or "false"
+   */
+  Real boolToReal(bool boolean_value) const;
+
+  /// Type of comparison to perform
+  const MooseEnum _comparison_type;
+
+  /// Absolute tolerance for "fuzzy" comparisons
+  const Real _absolute_tolerance;
+
+  /// Real value corresponding to "true"
+  static constexpr Real _real_true = 1.0;
+  /// Real value corresponding to "false"
+  static constexpr Real _real_false = -1.0;
```

Comment:
Yeah, I wondered if this would be odd. In general, I've seen both 0 and -1 used for this sort of purpose, but 0 is probably the more common... What's the preferred way of going back to boolean if anyone ever needed to do that?

Reply:
Ok, I didn't even realize there was a way to cast floating points to booleans... I had figured that it had no way to say that a value is exactly zero, but apparently the convention is that exact zero means false and everything else true, which seems risky. Anyway, I'll change it to 0.